diff -Naur '--from-file=ffmpeg-0.8.2' ffmpeg-0.8.2/ffmpeg-pmc.sh ffmpeg-0.8.2-psp/ffmpeg-pmc.sh
--- ffmpeg-0.8.2/ffmpeg-pmc.sh	1970-01-01 08:00:00 +0800
+++ ffmpeg-0.8.2-psp/ffmpeg-pmc.sh	2011-08-17 01:00:36 +0800
@@ -0,0 +1,14 @@
+./configure --prefix=C:/projects/music_player/ffmpeg-psp-0.8-music \
+--cross-prefix=psp- --enable-cross-compile --target-os=linux \
+--arch=mips --enable-hardcoded-tables --disable-fastdiv \
+--enable-gpl --disable-avfilter --disable-postproc \
+--disable-muxers --disable-encoders --disable-pthreads \
+--disable-protocols --disable-network --disable-aandct \
+--disable-indevs --disable-filters --disable-swscale \
+--disable-outdevs --disable-devices --disable-hwaccels \
+--disable-ffmpeg --disable-ffplay --disable-ffprobe \
+--disable-ffserver --disable-doc --disable-avdevice \
+--disable-vdpau --disable-dxva2 --disable-vaapi \
+--extra-libs='-lc -lpspnet -lpspnet_inet -lpspnet_apctl -lpspnet_resolver -lpsputility -lpspuser -lpspkernel' \
+--extra-cflags='-I/pspdev/psp/sdk/include -G0 -ffast-math -DO_BINARY=0' \
+--extra-ldflags='-L/pspdev/psp/sdk/lib'
\ No newline at end of file
diff -Naur '--from-file=ffmpeg-0.8.2' ffmpeg-0.8.2/libavformat/sol.c ffmpeg-0.8.2-psp/libavformat/sol.c
--- ffmpeg-0.8.2/libavformat/sol.c	2011-08-10 20:16:42 +0800
+++ ffmpeg-0.8.2-psp/libavformat/sol.c	2011-08-17 00:47:51 +0800
@@ -34,7 +34,7 @@
 {
     /* check file header */
     uint16_t magic;
-    magic=av_le2ne16(*((uint16_t*)p->buf));
+    magic=av_le2ne16((int16_t)(p->buf[1]<<8|p->buf[0]));
     if ((magic == 0x0B8D || magic == 0x0C0D || magic == 0x0C8D) &&
         p->buf[2] == 'S' && p->buf[3] == 'O' &&
         p->buf[4] == 'L' && p->buf[5] == 0)
diff -Naur '--from-file=ffmpeg-0.8.2' ffmpeg-0.8.2/libavutil/bswap.h ffmpeg-0.8.2-psp/libavutil/bswap.h
--- ffmpeg-0.8.2/libavutil/bswap.h	2011-08-10 20:15:52 +0800
+++ ffmpeg-0.8.2-psp/libavutil/bswap.h	2011-08-17 00:49:57 +0800
@@ -34,7 +34,9 @@
 
 #include "config.h"
 
-#if   ARCH_ARM
+#if   ARCH_MIPS
+#   include "mips/bswap.h"
+#elif ARCH_ARM
 #   include "arm/bswap.h"
 #elif ARCH_AVR32
 #   include "avr32/bswap.h"
diff -Naur '--from-file=ffmpeg-0.8.2' ffmpeg-0.8.2/libavutil/mips/bswap.h ffmpeg-0.8.2-psp/libavutil/mips/bswap.h
--- ffmpeg-0.8.2/libavutil/mips/bswap.h	1970-01-01 08:00:00 +0800
+++ ffmpeg-0.8.2-psp/libavutil/mips/bswap.h	2011-08-17 00:49:57 +0800
@@ -0,0 +1,45 @@
+/*
+ * 2011 RNB_PSP
+ *
+ * This file is part of FFmpeg.
+ *
+ * FFmpeg is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * FFmpeg is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with FFmpeg; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+/**
+ * @file
+ * byte swapping routines
+ */
+
+#ifndef AVUTIL_MIPS_BSWAP_H
+#define AVUTIL_MIPS_BSWAP_H
+
+#include <stdint.h>
+#include "config.h"
+#include "libavutil/attributes.h"
+
+#define av_bswap16 av_bswap16
+static av_always_inline av_const uint16_t av_bswap16(uint16_t x)
+{
+    return __builtin_allegrex_wsbh(x);
+}
+
+#define av_bswap32 av_bswap32
+static av_always_inline av_const uint32_t av_bswap32(uint32_t x)
+{
+    return __builtin_bswap32(x); // emits wsbw
+}
+
+#endif /* AVUTIL_MIPS_BSWAP_H */
diff -Naur '--from-file=ffmpeg-0.8.2' ffmpeg-0.8.2/libavformat/oma.c ffmpeg-0.8.2-psp/libavformat/oma.c
--- ffmpeg-0.8.2/libavformat/oma.c	2011-09-07 21:34:40 +0800
+++ ffmpeg-0.8.2-psp/libavformat/oma.c	2011-09-26 09:33:01 +0800
@@ -139,6 +139,16 @@
             framesize = ((codec_params & 0x3FF) * 8) + 8;
             st->codec->sample_rate = srate_tab[(codec_params >> 13) & 7]*100;
             st->codec->bit_rate    = st->codec->sample_rate * framesize * 8 / 1024;
+            
+            st->codec->extradata_size = 2;
+            edata = av_mallocz(2 + FF_INPUT_BUFFER_PADDING_SIZE);
+            if (!edata)
+                return AVERROR(ENOMEM);
+
+            st->codec->extradata = edata;
+            edata[0] = buf[0x22];
+            edata[1] = buf[0x23];
+
             av_set_pts_info(st, 64, 1, st->codec->sample_rate);
             av_log(s, AV_LOG_ERROR, "Unsupported codec ATRAC3+!\n");
             break;
diff -Naur '--from-file=ffmpeg-0.8.2' ffmpeg-0.8.2/libavformat/riff.c ffmpeg-0.8.2-psp/libavformat/riff.c
--- ffmpeg-0.8.2/libavformat/riff.c	2011-09-07 21:34:40 +0800
+++ ffmpeg-0.8.2-psp/libavformat/riff.c	2011-09-26 09:32:48 +0800
@@ -535,7 +535,20 @@
             }
             cbSize -= 22;
             size -= 22;
+            if (size>=4 && codec->codec_id == CODEC_ID_ATRAC3P)
+            {
+                avio_skip(pb, 2);
+                size -= 2;
+                cbSize = 2;
+            }
+        }
+        else if (id == 0xfffe && codec->codec_id == CODEC_ID_ATRAC3P && size >= 26)
+        {
+            avio_skip(pb, 24);
+            size -= 24;
+            cbSize = 2;
         }
+        
         codec->extradata_size = cbSize;
         if (cbSize > 0) {
             av_free(codec->extradata);
